swarm/api/routes/evolution.py

```
@@ -118,6 +120,21 @@ class PatchValidationResponse(BaseModel):
+    """Validate a path parameter and raise 400 if invalid."""
+    try:
+        return validate_path_component(value, name)
+    except ValueError as e:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "invalid_path_component",
+                "message": str(e),
+                "details": {"component": name, "value": value},
+            },
+        )
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)